e2etest/zt_newe2e_oauth_device_test.go

```diff
@@ -0,0 +1,118 @@
+
+		if !svm.Dryrun() {
+			// Make sure the LMT is in the past
+			time.Sleep(time.Second * 10)
+		}
+	}
+
+	body := NewRandomObjectContentContainer(svm, SizeFromString("10K"))
+	// Scale up from service to object
+	srcObj := CreateResource[ObjectResourceManager](svm, GetRootResource(svm, ResolveVariation(svm, []common.Location{common.ELocation.Local(), common.ELocation.Blob()})), ResourceDefinitionObject{
+		ObjectName: pointerTo("test"),
+		Body:       body,
+	})
+
+	// no local->local
+	if srcObj.Location().IsLocal() && dstObj.Location().IsLocal() {
+		svm.InvalidateScenario()
+		return
+	}
+	azcopyLogPathFolder := cmd.GetAzCopyAppPath()
+	azcopyJobPlanFolder := path.Join(azcopyLogPathFolder, "plans")
+
+	stdout, _ := RunAzCopy(
+		svm,
+		AzCopyCommand{
+			Verb: azCopyVerb,
+			Environment: &AzCopyEnvironment{
+				ManualLogin:     true,
+				LogLocation:     &azcopyLogPathFolder,
+				JobPlanLocation: &azcopyJobPlanFolder,
+			},
+			Targets: []ResourceManager{
+				TryApplySpecificAuthType(srcObj, EExplicitCredentialType.OAuth(), svm, CreateAzCopyTargetOptions{}),
+				TryApplySpecificAuthType(dstObj, EExplicitCredentialType.OAuth(), svm, CreateAzCopyTargetOptions{})},
+			Flags: CopyFlags{
+				CopySyncCommonFlags: CopySyncCommonFlags{
+					Recursive: pointerTo(true),
+				},
+			},
+		})
+
+	ValidateResource[ObjectResourceManager](svm, dstObj, ResourceDefinitionObject{
+		Body: body,
+	}, true)
+
+	ValidateAADAuth(svm, stdout, srcObj.Location(), dstObj.Location())
+}
+
+func ValidateAADAuth(a Asserter, stdout AzCopyStdout, src, dst common.Location) {
+	if dryrunner, ok := a.(DryrunAsserter); ok && dryrunner.Dryrun() {
+		return
+	}
+	// Check for successful login
+	loggedInDst := false
+	loggedInSrc := false
+	for _, p := range stdout.RawStdout() {
+		loggedInDst = loggedInDst || strings.Contains(p, "Authenticating to destination using Azure AD")
+		loggedInSrc = loggedInSrc || strings.Contains(p, "Authenticating to source using Azure AD")
+	}
+	a.AssertNow("source login should be successful", Equal{}, loggedInSrc, src.IsRemote())
+	a.AssertNow("dest login should be successful", Equal{}, loggedInDst, dst.IsRemote())
+}
+
+type DeviceLoginManualSuite struct {
+}
+
+func (s *DeviceLoginManualSuite) SetupSuite(a Asserter) {
+	stdout := RunAzCopyLoginLogout(a, AzCopyVerbLogin)
+	ValidateSuccessfulLogin(a, stdout)
+}
+
+func ValidateSuccessfulLogin(a Asserter, stdout AzCopyStdout) {
+	if dryrunner, ok := a.(DryrunAsserter); ok && dryrunner.Dryrun() {
+		return
+	}
+	// Check for successful login
+	loggedIn := false
+	for _, p := range stdout.RawStdout() {
+		loggedIn = loggedIn || strings.Contains(p, "Login succeeded")
+	}
+	a.AssertNow("login should be successful", Equal{}, loggedIn, true)
+}
+
+func (s *DeviceLoginManualSuite) TeardownSuite(a Asserter) {
+	RunAzCopyLoginLogout(a, AzCopyVerbLogout)
+}
+
+func (s *DeviceLoginManualSuite) Scenario_CopySync(svm *ScenarioVariationManager) {
+	Scenario_CopySync(svm, nil)
```

Comment:
ditto, this is odd.

Reply:
I did this so that we can eventually have more suites for the other OAuth methods, then we can just reference the one test and call it like this. But this might be overthinking/overplanning things

Reply:
This can easily be refactored during the cleanup PR, so I will leave this for now.